utils/sn_flux_est.py

```diff
@@ -438,83 +462,124 @@ def est_pulsar_flux(pulsar, obsid, plot_flux=False, metadata=None, query=None):
-        a = spind
-        a_err = spind_err
-
-        logger.debug("nu1 {0}".format(nu_1))
-        logger.debug("nu2 {0}".format(nu_2))
-        logger.debug("s2 {0}".format(s_2))
-        logger.info("calculating flux using spectral index: {0} and error: {1}".format(spind, spind_err))
-
-        flux_est = nu_1**a * nu_2**(-a) * s_2
-        #variance formula error est
-        s_2_var = nu_1**a * nu_2**(-a)
-        a_var = s_2 * nu_1**a * nu_2**(-a) * (np.log(nu_1)-np.log(nu_2))
-        a_var = a_var**2 * a_err**2
-        s_2_var = s_2_var**2 * s_2_err**2
-        flux_est_err = np.sqrt(s_2_var + a_var)
-
     elif len(flux_all) < 1:
-        logger.warning("No flux values on archive for {0}. Cannot estimate flux. Will return Nones".format(pulsar))
+        logger.warning("{} no flux values. Cannot estimate flux. Will return Nones".format(pulsar))
+
+    return spind, spind_err, K, covar_mat
+
+#---------------------------------------------------------------
+def est_pulsar_flux(pulsar, obsid, plot_flux=False, metadata=None, query=None):
+    """
+    Estimates a pulsar's flux from archival data by assuming a power law relation between flux and frequency
+
+    Parameters:
+    -----------
+    pulsar: string
+        The puslar's name. e.g. 'J2241-5236'
+    obsid: int
+        The observation ID
+    plot_flux: boolean
+        OPTIONAL - Whether or not to make a plot of the flux estimation. Default = False
+    metadata: list
+        OPTIONAL - The metadata call for this obsid
+    query: object
+        OPTIONAL - The return from psrqpy.QueryATNF for this pulsar
+
+    Returns:
+    -------
+    flux: float
+        The estimated flux in Jy
+    flux_err: float
+        The estimated flux's uncertainty in Jy
+    """
+
+    if metadata is None:
+        logger.debug("obtaining mean freq from obs metadata")
+        metadata = mwa_metadb_utils.get_common_obs_metadata(obsid)
+    f_mean = metadata[5]*1e6
+
+    freq_all, flux_all, flux_err_all, spind, spind_err = flux_from_atnf(pulsar, query=query)
+
+    logger.debug("Freqs: {0}".format(freq_all))
+    logger.debug("Fluxes: {0}".format(flux_all))
+    logger.debug("Flux Errors: {0}".format(flux_err_all))
+    logger.info("{0} there are {1} flux values available on the ATNF database"\
+                .format(pulsar, len(flux_all)))
+
+    if not spind or spind_err:
+        spind, spind_err, K, covar_mat = find_spind(pulsar, freq_all, flux_all, flux_err_all)
+
+    if K and covar_mat is not None and spind:
+        flux_est, flux_est_err = flux_from_plaw(f_mean, K, spind, covar_mat)
+    elif spind and spind_err:
+        flux_est, flux_est_err = flux_from_spind(f_mean, freq_all[0], flux_all[0], flux_err_all[0],\
+                                                spind, spind_err)
+    else:
+        logger.warning("{} no flux values on archive. Cannot estimate flux. Will return Nones".format(pulsar))
         return None, None
 
+    if plot_flux==True:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [comparison to True should be 'if cond is True:' or 'if cond:'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4980664)